src/bin/tikv-server.rs

```diff
@@ -181,24 +217,66 @@ fn main() {
         print_usage(&program, opts);
         return;
     }
-    initial_log(&matches);
 
-    let addr = matches.opt_str("A").unwrap_or_else(|| DEFAULT_LISTENING_ADDR.to_owned());
+    let config = match matches.opt_str("C") {
+        Some(path) => {
+            let mut config_file = fs::File::open(&path).expect("config open filed");
+            let mut s = String::new();
+            config_file.read_to_string(&mut s).expect("config read filed");
+            toml::Value::Table(toml::Parser::new(&s).parse().expect("malformed config file"))
+        }
+        // Empty value, lookup() always return `None`.
+        None => toml::Value::Integer(0),
+    };
+
+    initial_log(&matches, &config);
+
+    let addr = matches.opt_str("A").unwrap_or_else(|| {
```

Comment:
1. As far as I can see, we use `as_str` most of time. And we can pass `as_xxx` as a closure.
2. Use an `Option<T>` as default value can resolve it.